contrib/babelfishpg_unit/test_ssl_write.c

```diff
@@ -0,0 +1,108 @@
+#include "babelfishpg_unit.h"
+#include "../babelfishpg_tds/src/include/tds_secure.h"
+
+#ifdef USE_SSL
+static char *prelogin; 
+
+static ssize_t 
+mock_socket_write(Port *port, const void *ptr, size_t len)
+{
+
+    /*
+     *  Mock function of tds_secure_raw_write() present in tdssecure.c file
+     */
+
+    const unsigned char *ptr_u8 = (const unsigned char *) ptr;
+    signed char *prelogin_ptr = (signed char *) prelogin;
+    int i;
+    for (i = 0; i < len; i++) 
+        sscanf((const char *) &ptr_u8[i * 2], "%2hhx", (signed char *) &prelogin_ptr[i]);
+    return len;
+}
+
+
+TestResult*
+test_ssl_handshakeWrite(void)
+{
+
+    /*
+     *  We will generate a message and pass it to the handshake write function
+     *  We will be comparing the number of bytes written against the expected value
```

Comment:
Please update the comment saying we are also checking the actual bytes which are written

Reply:
Updated in the next revision.